tests/codegen/ccode/scripts/functions.py

```diff
@@ -0,0 +1,42 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=unused-variable
+
+def helloworld():
+    print('hello world')
+
+#$ header function incr(int)
+def incr(x):
+    x = x + 1
+
+#$ header function decr(int) results(int)
+def decr(x):
+    y = x - 1
+    return y
```

Comment:
great, you add more tests.
correct me if I am wrong, do we use the` result(type)` only for recursive functions.
I see this confusing even it just for tests, but the user sometimes reads the tests to know how to work with pyccel.

Reply:
no, it's required when writing a recursive function but it's optional for other function

Reply:
so maybe creating a recursive function test with the `result(type) `is more convenient.
so we let the uses of result type for its main purpose (as you know the programmer don't like too much typing).

Reply:
this codegen test was only in the fcode folder so I wake it available in the ccode, I think creating tests for functions is beyond the scope of this PR, I just didn't find where I can put test for a function with the `pass` statement so I copied the tests of the function from the fcode to the ccode

Reply:
For me I review the code that you added, I don't care about its source.
the important for me is to see the added changes clear and correctly.
But it's up to you if it's fine for you to copy-past inconvenient code a left it as it is, what could I say to you.

Reply:
here i think that a non-recursive function using a `result type` must also be tested, we can also add tests for recursive function

Reply:
@rakati is right that we shouldn't introduce poor/unexplainable code just because there is similar code elsewhere.
However I agree with @nhamidn here. The `result(type)` is optional for non-recursive functions, but that means that it shouldn't cause problems if it is specified so having 1 test for this is not a problem